### PR TITLE
added research slot

### PR DESCRIPTION
**Pull request by Buhle79**

### CODE REVIEW

File: base/actions/actions.py [hunk 1]
```diff
@@ -622,7 +643,11 @@ def get_healthcheck_data(self, tracker: Tracker, risk: Text) -> Dict[Text, Any]:
             "exposure": self.YES_NO_MAYBE_MAPPING[tracker.get_slot("exposure")],
             "tracing": self.YES_NO_MAPPING[tracker.get_slot("tracing")],
             "risk": risk,
+            "research_consent": self.YES_NO_MAPPING[
+                tracker.get_slot("research_consent")
+            ],
         }
+        print("Data: ", data)
```
Comment (erikh360) [inline base/actions/actions.py]: Remove print

File: base/actions/actions.py [hunk 2]
```diff
@@ -38,6 +38,10 @@ def yes_no_data(self) -> Dict[int, Text]:
     def yes_no_maybe_data(self) -> Dict[int, Text]:
         return {1: "yes", 2: "no", 3: "not sure"}
 
+    @property
+    def yes_no_more_data(self) -> Dict[int, Text]:
```
Comment (erikh360) [inline base/actions/actions.py]: Do we need this? We never save "more" or validate against it, we redirect before really validating

File: base/actions/actions.py [hunk 3]
```diff
@@ -176,7 +178,7 @@ def submit(
 class TBCheckProfileForm(BaseFormAction):
     """TBCheck form action"""
 
-    SLOTS = ["age"]
+    SLOTS = ["research_consent", "age"]
```
Comment (erikh360) [inline base/actions/actions.py]: Might need to collect age first if we want to skip research consent

File: base/actions/actions.py [hunk 4]
```diff
@@ -246,6 +245,12 @@ def slot_mappings(self) -> Dict[Text, Union[Dict, List[Dict]]]:
                 self.from_intent(intent="deny", value="no"),
                 self.from_text(),
             ],
+            "research_consent": [
+                self.from_intent(intent="affirm", value="yes"),
+                self.from_intent(intent="deny", value="no"),
+                self.from_intent(intent="more", value="more"),
```
Comment (erikh360) [inline base/actions/actions.py]: Is this more needed if we never save it?

File: base/tests/test_forms.py [hunk 5]
```diff
@@ -272,6 +289,7 @@ async def test_submit_to_healthconnect(self):
                 "city_location_coords": "+1.2-3.4",
                 "location_coords": "+3.4-1.2",
                 "location": "Cape Town, South Africa",
+                "research_consent": "no",
```
Comment (erikh360) [inline base/tests/test_forms.py:292]: I think the API is expecting boolean or None

Comment (erikh360) [reply base/tests/test_forms.py:292]: This applies to all payload tests below

**Review by erikh360 — COMMENTED** [review]
few small comment but looks good

File: base/actions/actions.py (hunk 2 above)
Comment (Buhle79) [reply base/actions/actions.py]: cool

File: base/tests/test_forms.py (hunk 5 above)
Comment (Buhle79) [reply base/tests/test_forms.py:292]: We mapping it on the payload, then i will send True or False

Comment (erikh360) [reply base/tests/test_forms.py:292]: I see, nice!

**Review by erikh360 — APPROVED** [review]
Nice!!

**Pull request merged**